File: js_project/Strings/RotationalPalindromeSlicer.js
let s1 = "radars";
let n = s1.length;
let length = n - 1;

let palindromeSliceFound = false;
let s2 = "";

for (let start = 0; start <= n - length; start++) {
    let s3 = "";
    for (let j = 0; j < length; j++) {
        s3 = s3 + s1.charAt(start + j);
        console.log(s3);
    }
    
    let left = 0;
    let right = s3.length - 1;
    let ispalendrome = true;
    
    while (left < right) {
        if (s3.charAt(left) !== s3.charAt(right)) {
            ispalendrome = false;
            break;
        }
        right--;
        left++;
    }
    
    if (ispalendrome) {
        palindromeSliceFound = true;
        s2 = s3;
        break;
    }
}

console.log("Original Text (" + n + " letters): " + s1);
console.log("Target Slice Size (N-1) : " + length);
console.log("----------------------------------------");
if (palindromeSliceFound) {
    console.log("Result: TRUE! Found a palindrome slice: \"" + s2 + "\"");
} else {
    console.log("Result: FALSE. No substring slice of length " + length + " is a palindrome.");
}

File: js_project/Strings/RotationalPalindromeSlicer.py
s1 = "radars"
n = len(s1)
length = n - 1

palindromeSliceFound = False
s2 = ""

for start in range(0, n - length + 1):
    s3 = ""
    for j in range(0, length):
        s3 = s3 + s1[start + j]
        print(s3)  
        
    left = 0
    right = len(s3) - 1
    ispalendrome = True
    
    while left < right:
        if s3[left] != s3[right]:
            ispalendrome = False
            break
        right = right - 1
        left = left + 1
        
    if ispalendrome:
        palindromeSliceFound = True
        s2 = s3
        break

print("Original Text (" + str(n) + " letters): " + s1)
print("Target Slice Size (N-1) : " + str(length))
print("----------------------------------------")
if palindromeSliceFound:
    print("Result: TRUE! Found a palindrome slice: \"" + s2 + "\"")
else:
    print("Result: FALSE. No substring slice of length " + str(length) + " is a palindrome.")